app/controllers/foreman_discovery/concerns/hosts_controller_extensions.rb

```diff
@@ -11,7 +11,7 @@ module HostsControllerExtensions
       # etc.. expect a params[:host] to work.
       def set_discovered_params
         return if params[:discovered_host].nil?
-        params[:host] ||= params[:discovered_host]
+        params[:host] ||= params.delete(:discovered_host)
```

Comment:
I found a case where both `host` and `discovered_host` were present, but fixed it in core https://github.com/theforeman/foreman/pull/5183/commits/d7d633db88844ba7349dfe6e46fae19b3bea006f